src/main/java/com/example/repick/domain/product/service/ProductService.java

```diff
@@ -207,9 +207,24 @@ public PageResponse<List<GetProductThumbnail>> getProducts(String type, ProductF
         User user = userRepository.findByProviderId(SecurityContextHolder.getContext().getAuthentication().getName())
                 .orElse(null);
         Long userId = user == null ? 0L : user.getId();  // non-login user 고려
+        List<String> sizes = productFilter.sizes();
+        if (productFilter.isMySize()) {
+            sizes = getUserSizesForCategory(user, Category.fromValue(productFilter.category()));
+        }
+        productFilter = new ProductFilter(
+                productFilter.keyword(),
+                productFilter.gender(),
+                productFilter.category(),
+                productFilter.styles(),
+                productFilter.minPrice(),
+                productFilter.maxPrice(),
+                productFilter.brandNames(),
+                productFilter.qualityRates(),
+                sizes,
+                productFilter.isMySize()
+        );
```

Comment:
```suggestion
            productFilter = new ProductFilter(
                    productFilter.keyword(),
                    productFilter.gender(),
                    productFilter.category(),
                    productFilter.styles(),
                    productFilter.minPrice(),
                    productFilter.maxPrice(),
                    productFilter.brandNames(),
                    productFilter.qualityRates(),
                    sizes,
                    productFilter.isMySize()
            );
        }
```

이렇게 바꾸면 MySize로 보는 사람의 경우에만 새로운 ProductFilter를 만들기 때문에 더 괜찮지 않을까요?

Reply:
감사합니다 ~ 수정했습니다 ㅎㅎ